### PR TITLE
add the ability to automaticly add custom javascript to dashboards

### PR DESCRIPTION
**Pull request by ReneTode**
creates a route for config/custom_javascript

adds all JS files from that route automaticly to the head includes

js files can already be included in the head by adding them to the skin, this option makes it possible to use js files in all skins.

### CODE REVIEW

**Review by Odianosen25 — CHANGES_REQUESTED** [review]

File: appdaemon/http.py [hunk 1]
```diff
@@ -790,6 +790,12 @@ def setup_dashboard_routes(self):
         if os.path.isdir(custom_css):
             self.app.router.add_static("/custom_css", custom_css)
 
+        # Add path for custom_javascript if it exists
+
+        custom_javascript = os.path.join(self.dashboard_obj.config_dir, "custom_javascript")
```
[the remark this answers is not in the document]
Comment (ReneTode) [reply appdaemon/http.py:795]: i just copied the lines that were there for custom_css

i expect that the check is done BEFORE setup_dashboard_routes is called.
the other change is in dashboard.py, so i guess there shouldnt be a check there either.
so i dont know where you want me to place a check

**Comment by Odianosen25** [comment]
Now we need to figure out why these changes are not passing checks

**Comment by frenck** [comment]
black...

**Comment by ReneTode** [comment]
i got no clue.

and black isnt telling me anything ;)

**Comment by frenck** [comment]
image

You need to format the code with black. Additionally, if you set up pre-commit correctly, this will be done on commit automatically (prevents you from running into these issues for the future).

**Comment by ReneTode** [comment]
> You need to format the code with black. Additionally, if you set up pre-commit correctly, this will be done on commit automatically (prevents you from running into these issues for the future).

i dont know how i format code with black, or how to setup pre-commit.

i just edited the files on github (in browser)

**Review by Odianosen25 — APPROVED** [review]

**Comment by Odianosen25** [comment]
@ReneTode, please can you add some docs for this?

Regards

**Comment by ReneTode** [comment]
yeah, ill do that tommorrow.

**Pull request merged**